### PR TITLE
feat: setting to always force push on save

### PR DESCRIPTION
**Pull request by lukecotter**
### What does this PR do?
Adds a new vscode setting to always use `--forceoverwrite` when doing `force:source:push` on save.
Useful if you do not care about conflicts or never make changes directly in an org
e.g prettier adds new line at end of a class which would result in a conflict on push 
### What issues does this PR fix or reference?
#2243

### Functionality Before
Push on save would just do a push meaning if conflicts existed the command with overwrite would need to be run.

### Functionality After
With the setting enabled it can be assumed that you local copy is correct and always overwrite on push on save.

### CODE REVIEW

File: packages/salesforcedx-vscode-core/src/commands/forceSourcePush.ts [hunk 1]
```diff
@@ -33,8 +34,11 @@ export class ForceSourcePushExecutor extends BaseDeployExecutor {
       .withArg('force:source:push')
       .withJson()
       .withLogName('force_source_push_default_scratch_org');
-    if (this.flag === '--forceoverwrite') {
-      builder.withArg(this.flag);
+    if (
+      sfdxCoreSettings.getForcePushAndPullEnabled() ||
+      this.flag === '--forceoverwrite'
+    ) {
+      builder.withArg('--forceoverwrite');
```
Comment (brpowell) [inline packages/salesforcedx-vscode-core/src/commands/forceSourcePush.ts:41]: Might want to enable the force push here instead:

https://github.com/forcedotcom/salesforcedx-vscode/blob/06e7e5fa1b8ef4c19b2057d56d679003a3ccbcba/packages/salesforcedx-vscode-core/src/settings/pushOrDeployOnSave.ts#L84

and have a check like

```typescript
let command = 'sfdx.force.source.push';
if (sfdxCoreSettings.getForcePushOnSave()) {
  command += '.force'
}
vscode.commands.executeCommand(command);
```

**Comment by brpowell** [comment]
Thanks for contributing @lukecotter 💯 . For `pull` I'm not sure we want to have a setting to force overriding conflicts, since we already a command for both `pull` and `force pull`. With `push` though this makes sense with the deploy on save scenario - we just call the force push version of the command instead since like pull, push has two versions. I left a comment on where to go about making that change.

File: packages/salesforcedx-vscode-core/package.json [hunk 2]
```diff
@@ -845,6 +845,11 @@
       "type": "object",
       "title": "%feature_previews_title%",
       "properties": {
+        "salesforcedx-vscode-core.force-push-and-pull.enabled": {
```
Comment (brpowell) [inline packages/salesforcedx-vscode-core/package.json:848]: @smaddox-sf Looking for your feedback on the setting name. Following my comment on how this is a deploy on save feature, should we make the setting in that scope?

e.g.

`salesforcedx-vscode-core.push-or-deploy-on-save.enabled`
`salesforcedx-vscode-core.push-or-deploy-on-save.overrideConflictsOnPush`

**Comment by lukecotter** [comment]
@brpowell I had originally done it that way. Lol.
I'll make the change to be push only, I agree that it makes sense when push on save is on.

File: packages/salesforcedx-vscode-core/package.json (hunk 2 above)
Comment (smaddox-sf) [reply packages/salesforcedx-vscode-core/package.json:848]: @brpowell - I like your suggestion and agree we should keep it within the 'upon save' scope.  For the description of this setting, we should also note it's upon save and remove the mention of pull...so something like "Specifies whether to always use --forceoverwrite when doing force:source:push on save".  When ready, Sonal should review the final wording for the setting & description

**Comment by lukecotter** [comment]
@brpowell This should be done now.
Let me know if there is anything else, needed.

File: packages/salesforcedx-vscode-core/src/commands/forceSourcePush.ts (hunk 1 above)
Comment (lukecotter) [reply packages/salesforcedx-vscode-core/src/commands/forceSourcePush.ts]: done

File: packages/salesforcedx-vscode-core/package.json (hunk 2 above)
Comment (lukecotter) [reply packages/salesforcedx-vscode-core/package.json]: I have used the suggested text and the setting is called `salesforcedx-vscode-core.push-or-deploy-on-save`
Let me know if this needs to change

File: packages/salesforcedx-vscode-core/package.json [hunk 3]
```diff
@@ -837,6 +837,11 @@
       "type": "object",
       "title": "%feature_previews_title%",
       "properties": {
+        "salesforcedx-vscode-core.force-push-on-save": {
```
Comment (brpowell) [inline packages/salesforcedx-vscode-core/package.json:840]:
```suggestion
        "salesforcedx-vscode-core.push-or-deploy-on-save.overrideConflictsOnPush": {
```

Can we move this setting right below the `push-or-deploy-on-save.enabled` entry to group related setting?

@smaddox-sf @sbudhirajadoc what do you think of my suggested setting name (Override conflicts on push)? Also moved the setting into the `push-or-deploy-on-save` namespace.

File: packages/salesforcedx-vscode-core/package.nls.json [hunk 4]
```diff
@@ -37,6 +37,7 @@
   "force_project_create_text": "SFDX: Create Project",
   "force_project_with_manifest_create_text": "SFDX: Create Project with Manifest",
   "force_apex_trigger_create_text": "SFDX: Create Apex Trigger",
+  "force_push_on_save_description": "Specifies whether to always use --forceoverwrite when doing force:source:push on save.",
```
Comment (brpowell) [inline packages/salesforcedx-vscode-core/package.nls.json:40]: @sbudhirajadoc

Comment (brpowell) [inline packages/salesforcedx-vscode-core/package.nls.json:40]:
```suggestion
  "override_conflicts_on_push_description": "Specifies whether to always use --forceoverwrite when doing force:source:push on save.",
```

Comment (lukecotter) [reply packages/salesforcedx-vscode-core/package.nls.json]: done

File: packages/salesforcedx-vscode-core/package.json (hunk 3 above)
Comment (lukecotter) [reply packages/salesforcedx-vscode-core/package.json:840]: done

Comment (smaddox-sf) [reply packages/salesforcedx-vscode-core/package.json:840]: Looks good on both.  @sbudhirajadoc will do a final review on the setting name & description

File: packages/salesforcedx-vscode-core/package.json (hunk 2 above)
Comment (sbudhirajadoc) [reply packages/salesforcedx-vscode-core/package.json]: Slight modification -- "Specifies whether to always use --forceoverwrite when you run force:source:push on save"

File: packages/salesforcedx-vscode-core/package.nls.json (hunk 4 above)
Comment (sbudhirajadoc) [reply packages/salesforcedx-vscode-core/package.nls.json]: Slight modification -- "Specifies whether to always use --forceoverwrite when you run force:source:push on save"

File: packages/salesforcedx-vscode-core/package.json (hunk 3 above)
Comment (sbudhirajadoc) [reply packages/salesforcedx-vscode-core/package.json:840]: added a small update on description. setting name is good.

File: packages/salesforcedx-vscode-core/package.nls.json (hunk 4 above)
Comment (sbudhirajadoc) [reply packages/salesforcedx-vscode-core/package.nls.json]: Slight modification -- "Specifies whether to always use --forceoverwrite when you run force:source:push on save"

File: packages/salesforcedx-vscode-core/package.json [hunk 5]
```diff
@@ -837,6 +837,11 @@
...
+          "type": "boolean",
+          "default": false,
+          "description": "%force_push_on_save_description%"
```
Comment (brpowell) [inline packages/salesforcedx-vscode-core/package.json:843]:
```suggestion
          "description": "%override_conflicts_on_push_description%"
```

File: packages/salesforcedx-vscode-core/package.nls.json [hunk 6]
```diff
@@ -44,6 +44,7 @@
   "retrieve_test_code_coverage_text": "Specifies whether code coverage results are calculated and retrieved when you run Apex tests.",
   "telemetry_enabled_description": "Specifies whether to enable Salesforce telemetry. Even when enabled, still abides by the overall `telemetry.enableTelemetry` vscode setting.",
   "push_or_deploy_on_save_enabled_description": "Specifies whether or not to automatically run force:source:push (for source-tracked orgs) or force:source:deploy (for non-source-tracked orgs) when a local source file is saved.",
+  "override_conflicts_on_push_description": "Specifies whether to always use --forceoverwrite when doing force:source:push on save.",
```
Comment (brpowell) [inline packages/salesforcedx-vscode-core/package.nls.json:47]:
```suggestion
  "override_conflicts_on_push_description": "Specifies whether to always use --forceoverwrite when you run force:source:push on save",
```

**Comment by lukecotter** [comment]
@brpowell is this ok to merge now?

**Review by brpowell — APPROVED** [review]

**Pull request merged**